src/shogun/lib/any.h

```diff
@@ -170,17 +197,149 @@ namespace shogun {
 		virtual ~AnyVisitor() = default;
 
 		virtual void on(bool*) = 0;
+		virtual void on(char*) = 0;
+		virtual void on(int8_t*) = 0;
+		virtual void on(uint8_t*) = 0;
+		virtual void on(int16_t*) = 0;
+		virtual void on(uint16_t*) = 0;
 		virtual void on(int32_t*) = 0;
+		virtual void on(uint32_t*) = 0;
 		virtual void on(int64_t*) = 0;
-		virtual void on(float*) = 0;
-		virtual void on(double*) = 0;
+		virtual void on(uint64_t*) = 0;
+		virtual void on(float32_t*) = 0;
+		virtual void on(float64_t*) = 0;
+		virtual void on(floatmax_t*) = 0;
+		virtual void on(complex128_t*) = 0;
 		virtual void on(CSGObject**) = 0;
-		virtual void on(SGVector<int>*) = 0;
-		virtual void on(SGVector<float>*) = 0;
-		virtual void on(SGVector<double>*) = 0;
-		virtual void on(SGMatrix<int>*) = 0;
-		virtual void on(SGMatrix<float>*) = 0;
-		virtual void on(SGMatrix<double>*) = 0;
+		virtual void enter_matrix(index_t* rows, index_t* cols) = 0;
```

Comment:
@lisitsyn this is what you had in mind?

Reply:
yeah something like that

I also thought of something like
`virtual AnyVisitor* enter_matrix(index_t* rows, index_t* cols) = 0;`
So that it returns a new sub-visitor.